### PR TITLE
Update ouroboros-network dependency

### PR DESCRIPTION
**Pull request by mrBliss**
We switch to `initChainDB` instead of manually constructing the `ChainDbArgs`, it is easier and more future-proof.

@karknu made the network-related changes.

### CODE REVIEW

**Comment by karknu** [comment]
> TODO: because of network changes this doesn't build yet. @karknu Can you take care of this?

Sure, mind if I push to this branch?

**Comment by mrBliss** [comment]
> > TODO: because of network changes this doesn't build yet. @karknu Can you take care of this?
> 
> Sure, mind if I push to this branch?

Go ahead! Thanks

File: src/exec/Shelley.hs [hunk 1]
```diff
@@ -94,7 +82,7 @@ mkParams rr cdb nconf nstate blockchainTime = NodeArgs
   , chainDB = cdb
   , callbacks = NodeCallbacks
       { produceDRG = drgNew
-      , produceBlock = \_ _ _ _ _ _ -> error "cannot produce blocks, so why do I have to give this?"
+      , produceBlock = \_ _ _ _ _ _ -> error "ZXXcannot produce blocks, so why do I have to give this?"
```
Comment (mrBliss) [inline src/exec/Shelley.hs]: Oops?

Comment (karknu) [reply src/exec/Shelley.hs]: Yeah, looks like fat fingers.

**Comment by mrBliss** [comment]
Thanks for the help, @karknu :pray:

**Review by dcoutts — APPROVED** [review]

**Pull request merged**